src/d3a/models/powerflow/__init__.py

```diff
@@ -0,0 +1,132 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import pandapower as pp
+from pandapower.plotting import to_html
+
+from d3a.models.strategy.load_hours import LoadHoursStrategy
+from d3a.models.strategy.storage import StorageStrategy
+from d3a.models.strategy.pv import PVStrategy
+from d3a.models.strategy.finite_power_plant import FinitePowerPlant
+from d3a.models.strategy.infinite_bus import InfiniteBusStrategy
+
+
+class PowerFlow:
+    def __init__(self, root_area, root_voltage=400):
+        """
+        :param root_area: It contains tree structured hierarchical energy grid
+        :param root_voltage: It contains the voltage of the top most hierarchy
+        """
+        self.network = pp.create_empty_network()
+        self.root_voltage = root_voltage
+        self._grid_forming(root_area)
+
+    def _grid_forming(self, area):
+        """
+        :param area: Takes area object and convert it to specific energy device
+        :return: It will return pandapower compliant connected grid
+        """
+        if area.strategy is None:
+            area.bus = None
+            area.bus = self._create_bus(area, self.root_voltage)
+            if area.parent is not None:
+                area.line = None
```

Comment:
Do we really need these lines here? Can you please explain why. Thanks

Reply:
You are right, removing it.